presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusClient.java

```
@@ -151,10 +162,52 @@ public PrometheusTable getTable(String schema, String tableName)
     private static Map<String, Object> lookupMetrics(URI metadataUri, JsonCodec<Map<String, Object>> metricsCodec)
             throws IOException
     {
-        URL result = metadataUri.toURL();
-        String json = Resources.toString(result, UTF_8);
+        String json = getHttpResponse(metadataUri).string();
         Map<String, Object> metrics = metricsCodec.fromJson(json);
 
         return metrics;
     }
+
+    static ResponseBody getHttpResponse(URI uri)
+            throws IOException
+    {
+        Request.Builder requestBuilder = new Request.Builder();
+        getBearerAuthInfoFromFile().map(bearerToken ->
+                requestBuilder.header("Authorization", "Bearer " + bearerToken));
+        requestBuilder.url(uri.toURL());
+        Request request = requestBuilder.build();
+        Response response = httpClient.newCall(request).execute();
+        if (response.isSuccessful()) {
+            return response.body();
+        }
+        else {
+            throw new PrestoException(REMOTE_TASK_ERROR, "Bad response " + response.code() + response.message());
+        }
+    }
+
+    static Optional<String> getBearerAuthInfoFromFile()
+    {
+        String tokenFileName = config.getBearerTokenFile();
+        if (tokenFileName != null) {
+            DataInputStream reader = null;
+            try {
+                File tokenFile = new File(tokenFileName);
+                return Optional.of(Files.toString(tokenFile, Charsets.UTF_8));
+            }
+            catch (Exception e) {
+                throw new PrestoException(NOT_FOUND, "Failed to find/read file: " + tokenFileName, e);
+            }
+            finally {
+                try {
+                    if (reader != null) {
+                        reader.close();
+                    }
+                }
+                catch (IOException e) {
+                    //give up quietly
+                }
+            }
+        }
+        return Optional.empty();
```

I think we should fail if the token file is configured but not found.